packages/contracts-bedrock/test/vendor/eas/SchemaRegistry.t.sol

```diff
@@ -0,0 +1,512 @@
+
+        assertEq(record.uid, bytes32(0));
+        assertEq(record.schema, "");
+        assertEq(address(record.resolver), address(0));
+        assertEq(record.revocable, false);
+    }
+
+    // =============================================================
+    //                      ERROR CASE TESTS
+    // =============================================================
+    /// @dev Tests duplicate schema registration prevention.
+    ///      1. Setup:
+    ///         - Creates simple boolean schema
+    ///         - Uses zero address for resolver
+    ///         - Enables revocation
+    ///      2. First Registration:
+    ///         - Successfully registers schema
+    ///      3. Duplicate Attempt:
+    ///         - Attempts to register same schema again
+    ///         - Verifies revert with AlreadyExists error
+    ///      Ensures system properly prevents duplicate
+    ///      schema registrations, maintaining schema uniqueness
+    function testCannotRegisterSameSchemaTwice() public {
+        string memory schema = "bool isFriend";
+        address resolver = address(0);
+        bool revocable = true;
+
+        registry.register(schema, ISchemaResolver(resolver), revocable);
+
+        vm.expectRevert(abi.encodeWithSignature("AlreadyExists()"));
+        registry.register(schema, ISchemaResolver(resolver), revocable);
+    }
+
+    // =============================================================
+    //                      EVENT TESTS
+    // =============================================================
+    /// @dev Tests schema registration event emission.
+    ///      1. Setup:
+    ///         - Creates basic boolean schema
+    ///         - Sets specific resolver address
+    ///         - Enables revocation
+    ///         - Calculates expected UID
+    ///      2. Event Testing:
+    ///         - Creates expected schema record
+    ///         - Sets up event emission expectation
+    ///         - Registers schema
+    ///      3. Verification:
+    ///         - Verifies Registered event emission
+    ///         - Validates all schema parameters:
+    ///           * UID matches expected
+    ///           * Schema string correct
+    ///           * Resolver address matches
+    ///           * Revocable flag set properly
+    ///      Ensures proper event emission and data accuracy
+    ///      during schema registration
+    function testRegisterSchemaEvent() public {
+        string memory schema = "bool like";
+        address resolver = address(0x123);
+        bool revocable = true;
+
+        bytes32 expectedUID = _getUID(schema, resolver, revocable);
+        SchemaRecord memory expectedSchema = SchemaRecord({
+            uid: expectedUID,
+            schema: schema,
+            resolver: ISchemaResolver(resolver),
+            revocable: revocable
+        });
+
+        vm.expectEmit(true, true, true, true, address(registry));
```

Comment:
smol nit: you can do `vm.expectEmit(address(registry))` and its the same thing as using all of the true values, less complicated that way